File: Pratica 02/Sorveteria/C++/sorveteria.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int perguntaQuantidade(){
    int quantidade;
    cout << "Quantidade deste produto em estoque: ";
    cin >> quantidade;
    // verifica se é negativo ou não
    if(quantidade < 0){
        cout << "Erro. A quantidade não pode ser um valor negativo. Por favor, tente novamente." << endl;

        return perguntaQuantidade();
    }

    return quantidade;
}

int main(){
    // Declaração variáveis
    string nome_produto;
    int quantidade_produto;
    float preco_unitario = 0;

    // Entradas
    cout << "Informe o nome do produto: "; cin >> nome_produto;
    quantidade_produto = perguntaQuantidade();
    cout << "Informe o preço unitário: "; cin >> preco_unitario;

    // Saídas
    cout << " ----- Informações do produto -----" << endl;
    cout << "Nome: " << nome_produto << endl;
    cout << "Quantidade em estoque: " << quantidade_produto << endl;
    cout << "Preço unitário: R$" << fixed << setprecision(2) << preco_unitario << endl;

    return 0;
}

File: Pratica 02/Sorveteria/Pyhton/sorveteria.py
# Regra: quantidade em estoque não pode ser negativa

# funçao que pergunta e valida quantidade em estoque
def perguntaQuantidade():

    # Solicitação (entrada)
    quantidade = int(input("Quantidade do produto em estoque: "))

    # Validação
    if(quantidade< 0):
        print("Erro: A quantidade não pode ser um valor negativo. Por favor, tente novamente.")
        perguntaQuantidade();

    return quantidade

# declaração variáveis
nome_produto = str(input("Digite o nome do produto: "))
quantidade_produto = perguntaQuantidade()
preco_unitario = float(input("Digite o preço do produto: "))

# saída
print()
print("-=" * 13)
print(f"""Nome: {nome_produto}
Quantidade em estoque: {quantidade_produto}
Preço Unitário: R${preco_unitario:.2f}""")
print("-=" * 13)